WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/EditorPage.java

```diff
@@ -5,52 +5,100 @@
+import static android.support.test.espresso.Espresso.openActionBarOverflowOrOptionsMenu;
+import static android.support.test.espresso.Espresso.pressBack;
 import static android.support.test.espresso.action.ViewActions.click;
 import static android.support.test.espresso.action.ViewActions.typeText;
 import static android.support.test.espresso.assertion.ViewAssertions.matches;
 import static android.support.test.espresso.matcher.RootMatchers.isDialog;
-import static android.support.test.espresso.matcher.ViewMatchers.isDescendantOfA;
 import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
 import static android.support.test.espresso.matcher.ViewMatchers.withHint;
 import static android.support.test.espresso.matcher.ViewMatchers.withId;
 import static android.support.test.espresso.matcher.ViewMatchers.withText;
 import static org.hamcrest.Matchers.allOf;
+import static org.wordpress.android.support.WPSupportUtils.clickOn;
 import static org.wordpress.android.support.WPSupportUtils.isElementDisplayed;
+import static org.wordpress.android.support.WPSupportUtils.populateTextField;
 import static org.wordpress.android.support.WPSupportUtils.waitForElementToBeDisplayed;
+import static org.wordpress.android.support.WPSupportUtils.withIndex;
 
 public class EditorPage {
     private static ViewInteraction publishButton = onView(withId(R.id.menu_save_post));
-
     private static ViewInteraction editor = onView(withId(R.id.aztec));
-
-    private static ViewInteraction titleField = onView(allOf(withId(R.id.title), withHint("Title")));
-
+    private static ViewInteraction titleField = onView(allOf(withId(R.id.title),
+            withHint("Title")));
     private static ViewInteraction publishConfirmation = onView(allOf(
             withText("Post published"), isDescendantOfA(withId(R.id.snackbar))));
+    private static ViewInteraction addMediaButton = onView(withId(R.id.media_button_container));
+    private static ViewInteraction mediaLibraryButton = onView(withId(R.id.media_bar_button_library));
+    private static ViewInteraction allowMediaAccessButton = onView(allOf(withId(R.id.button),
+            withText("Allow")));
+    private static ViewInteraction confirmButton = onView(withId(R.id.mnu_confirm_selection));
 
     public EditorPage() {
         editor.check(matches(isDisplayed()));
     }
 
-    public EditorPage enterTitle(String postTitle) {
+    public void enterTitle(String postTitle) {
         titleField.perform(typeText(postTitle), ViewActions.closeSoftKeyboard());
-
-        return this;
     }
 
-    public EditorPage enterContent(String postContent) {
+    public void enterContent(String postContent) {
         editor.perform(typeText(postContent), ViewActions.closeSoftKeyboard());
+    }
+
+    // Image needs a little time to be uploaded after entering the image
+    public void enterImage() {
+        // Click on add media button
+        clickOn(addMediaButton);
```

Comment:
I'm getting a failure at `testPublishFullPost` at this step because the test is doing a long press on the media icon instead of a regular tap. I remember this happening when I worked on Espresso tests in the past but I can't remember if there's a way to avoid it; have you seen this happen before?

Reply:
Ahh I'm trying to find a better way for this but for now the only workaround is `adb shell settings put secure long_press_timeout 1500` This is an issue that's known and noted in the Espresso code https://android.googlesource.com/platform/frameworks/testing/+/android-support-test/espresso/core/src/main/java/android/support/test/espresso/action/GeneralClickAction.java#75

Reply:
I updated the PR description to mention this in the meantime

Reply:
We could create a gradle task for running the tests and run that adb command as a pre-step.. sounds good?

Reply:
Ah yes, that's it! Running that command resolved it for me, and adding it as a pre-step sounds like a good plan to ensure the click action is more reliable for these tests.

Reply:
An `adb` command won't necessarily work as a solution to this. We want to be able to run these tests on Firebase Test Lab and we don't have access to configure those devices with `adb`.

Is there any possible Android/Espresso workaround for this.

Reply:
@jtreanor The usual workaround is to use Espresso's Rollback action to undo anything caused by the long press but in our case this wouldn't work 😕  I'm checking if maybe there's a way to workaround this with UiAutomator but if not I can look into if disabling the long press action on the button is an option.

Reply:
https://github.com/wordpress-mobile/WordPress-Android/pull/9795/files#diff-b3917ad698df47a8f8ddd7d3b8315e09R56

@jtreanor @rachelmcr this seems to do the trick!

Reply:
Nice find @JavonDavis! This is working for me now,.